test/integration/region_aware_routing_test.exs

```diff
@@ -0,0 +1,166 @@
+defmodule Realtime.Integration.RegionAwareRoutingTest do
+  use Realtime.DataCase, async: false
+  use Mimic
+
+  alias Realtime.Api
+  alias Realtime.Api.Tenant
+
+  setup do
+    Ecto.Adapters.SQL.Sandbox.mode(Realtime.Repo, {:shared, self()})
+
+    # Configure test runner as non-master region (eu-west-1) with master_region = us-east-1
+    original_region = Application.get_env(:realtime, :region)
+    original_master_region = Application.get_env(:realtime, :master_region)
+
+    Application.put_env(:realtime, :region, "eu-west-1")
+    Application.put_env(:realtime, :master_region, "us-east-1")
+
+    local_pid =
+      :syn.members(RegionNodes, "us-east-1")
+      |> Enum.find_value(fn {pid, _} -> if node(pid) == node(), do: pid end)
+
+    if local_pid do
+      :syn.leave(RegionNodes, "us-east-1", local_pid)
+      :syn.join(RegionNodes, "eu-west-1", local_pid, node: node())
+    end
+
+    # Start peer node as master region (us-east-1)
+    # The master node will automatically register itself in RegionNodes on startup
+    {:ok, master_node} =
+      Clustered.start(nil,
+        extra_config: [
+          {:realtime, :region, "us-east-1"},
+          {:realtime, :master_region, "us-east-1"}
+        ]
+      )
```

Comment:
I'm just trying to understand this whole setup so sorry for any stupid questions!

Could we instead of changing the local region and the syn region for the main/local node just:

* Set the `master_region` to `eu-west-1`
* Start the new remote node with region and master_region set to `eu-west-1`

This way you don't need to change the local node that is running. Less things to go wrong I think? No need to syn.leave and syn.join twice and no need to change the `region`

The only reason why I am asking is that changing region and messing with syn RegionNodes for the main node can cause flakyness.

WDYT?